faust/web/blueprints.py

```diff
@@ -0,0 +1,117 @@
+from pathlib import Path
+import typing
+from typing import List, NamedTuple, Optional, Type, Union, cast
+
+from faust.types import AppT
+from faust.types.web import BlueprintT, PageArg, RouteDecoratorRet, View, Web
+
+if typing.TYPE_CHECKING:
+    from faust.app import App
+else:
+    class App: ...  # noqa
+
+__all__ = ['Blueprint']
+
+
+class FutureRoute(NamedTuple):
+    uri: str
+    name: Optional[str]
+    handler: PageArg
+    base: Type[View]
+
+
+class FutureStaticRoute(NamedTuple):
+    uri: str
+    file_or_directory: Path
+    name: str
+
+
+class Blueprint(BlueprintT):
+    routes: List[FutureRoute]
+    static_routes: List[FutureStaticRoute]
+    view_name_separator: str = ':'
+
+    def __init__(self,
+                 name: str,
+                 *,
+                 url_prefix: Optional[str] = None) -> None:
+        self.name = name
+        self.url_prefix = url_prefix
+
+        self.routes = []
+        self.static_routes = []
+
+    def clone(self, url_prefix: Optional[str] = None) -> BlueprintT:
+        if url_prefix is None:
+            url_prefix = self.url_prefix
+        bp = type(self)(name=self.name, url_prefix=url_prefix)
+        bp.routes = self.routes  # XXX Do not modify!!!
```

Comment:
Maybe make the comment more explicit. This is confusing as I though `bp.routes` should never be modified but adding more routes to the blueprint will update this.